application/event_processor/dos.py

```diff
@@ -215,12 +217,24 @@ def _connect_dos_db() -> None:
         password=db_password,
         connect_timeout=30,
         options=f"-c search_path=dbo,{db_schema}",
+        application_name=f"DI-event-processor <psycopg2> tid={trace_id}"
     )
 
     return db
 
 
-def query_dos_db(sql_command: str) -> DictCursor:
+def disconnect_dos_db() -> None:
+    """Closes the DoS database connection if it exists and is open"""
+    global db_connection
+    if db_connection is not None:
+        try:
+            db_connection.close()
+            logger.info("The DoS database connection was closed.")
+        except Exception as e:
+            logger.error(f"There was an exception while trying to close DoS database connection: {e}")
```

Comment:
Can this be logger.exception as it shows both a logger.error and prints the stack trace